src/main/java/com/openshift/restclient/model/volume/IPersistentVolume.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.openshift.restclient.model.volume;
+
+import java.util.Set;
+
+import com.openshift.internal.restclient.model.volume.NFSVolume;
+import com.openshift.restclient.model.IResource;
+import com.openshift.restclient.utils.MemoryUnit;
+
+
+public interface IPersistentVolume extends IResource {
+
+	long getCapacity(String unit);
+	long getCapacity(MemoryUnit unit);
+	long getCapacity();
+	MemoryUnit getCapacityUnit();
+	void setCapacity(long capacity, MemoryUnit unit);
+
+	Set<String> getAccessModes();
+	void setAccessModes(String... modes);
+
+	String getReclaimPolicy();
+	void setReclaimPolicy(String policy);
+
+	NFSVolume getNFSVolume();
+	/**
+	 * @see https://docs.openshift.com/enterprise/3.0/rest_api/kubernetes_v1.html#v1-nfsvolumesource
+	 */
+	void setNFSVolume(NFSVolume volume);
+
+	String getHostPathVolume();
```

Comment:
HostPath is actually one of kubernetes resources. I will add common method getPath to IVolume and rewrite getVolume to be more generic.